### PR TITLE
IGNITE-21303 Fix the way we save new assignments for "resetPartitions" command

### PR DESCRIPTION
**Pull request by ibessonov**
https://issues.apache.org/jira/browse/IGNITE-21303

Thank you for submitting the pull request.

To streamline the review process of the patch and ensure better code quality
we ask both an author and a reviewer to verify the following:

### The Review Checklist
- [ ] **Formal criteria:** TC status, codestyle, mandatory documentation. Also make sure to complete the following:  
\- There is a single JIRA ticket related to the pull request.  
\- The web-link to the pull request is attached to the JIRA ticket.  
\- The JIRA ticket has the Patch Available state.  
\- The description of the JIRA ticket explains WHAT was made, WHY and HOW.  
\- The pull request title is treated as the final commit message. The following pattern must be used: IGNITE-XXXX Change summary where XXXX - number of JIRA issue.
- [ ] **Design:** new code conforms with the design principles of the components it is added to.
- [ ] **Patch quality:** patch cannot be split into smaller pieces, its size must be reasonable.
- [ ] **Code quality:** code is clean and readable, necessary developer documentation is added if needed.
- [ ] **Tests code quality:** test set covers positive/negative scenarios, happy/edge cases. Tests are effective in terms of execution time and resources.

### Notes
- [Apache Ignite Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Java+Code+Style+Guide)

### CODE REVIEW

File: modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java [hunk 1]
```diff
@@ -1835,9 +1834,7 @@ private CompletableFuture<Void> handleChangePendingAssignmentEvent(
             // In case of forced assignments we need to remove nodes that are present in the stable set but are missing from the
             // pending set. Such operation removes dead stable nodes from the resulting stable set, which guarantees that we will
```
Comment (ibessonov) [inline modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java:1835]: I guess I should change the comment

Comment (tkalkirill) [reply modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java:1835]: I don't think so, but you can fix it later.

**Review by tkalkirill — APPROVED** [review]

File: modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java (hunk 1 above)
Comment (sanpwc) [reply modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java:1837]: I believe that comment should be changed, currently it seems misleading.

File: modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java [hunk 2]
```diff
@@ -1877,12 +1874,12 @@ private CompletableFuture<Void> handleChangePendingAssignmentEvent(
                                 isRecovery
                         );
                     }), ioExecutor);
-        } else {
+        } else if (pendingAssignmentsAreForced && localMemberAssignment != null) {
             localServicesStartFuture = runAsync(() -> {
-                if (pendingAssignmentsAreForced && replicaMgr.isReplicaStarted(replicaGrpId)) {
-                    replicaMgr.resetPeers(replicaGrpId, fromAssignments(computedStableAssignments.nodes()));
-                }
+                replicaMgr.resetPeers(replicaGrpId, fromAssignments(computedStableAssignments.nodes()));
```
Comment (sanpwc) [inline modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java]: BusyLock is missing.
I've added it while adding changes to the PartitionReplicaLifecycleManager. Feel free to revert if you don't agree.

**Review by sanpwc — APPROVED** [review]

**Pull request merged**